skunk/src/io.github.iltotore.iron/skunk.scala

```diff
@@ -0,0 +1,33 @@
+package io.github.iltotore.iron
+
+import _root_.skunk.*
+
+import scala.Conversion
+
+object skunk:
+
+  /**
+   * Explicit conversion for refining a [[Codec]]. Decodes to the underlying type then checks the constraint.
+   *
+   * @param constraint the [[Constraint]] implementation to test the decoded value
+   */
+  extension [A](codec: Codec[A])
+    inline def refined[C](using inline constraint: Constraint[A, C]): Codec[A :| C] =
+      codec.eimap[A :| C](_.refineEither[C])(_.asInstanceOf[A])
+
+  /**
+   * Implicit conversion for refining a [[Codec]]. Decodes to the underlying type then checks the constraint.
+   *
+   * @param constraint the [[Constraint]] implementation to test the decoded value
+   */
+  inline given [A, C](using inline constraint: Constraint[A, C]): Conversion[Codec[A], Codec[A :| C]] =
```

Comment:
I don't think such conversion should be included. It adds magic and complexity. Furthermore, doing `codec.refined[Constraint]` or summoning `Codec[A :| C]` does not seem painful to me.

Also, note that similar Iron modules (ZIO JSON/Circe/Doobie/... support) do not include such constraint.

Reply:
Agreed, I got rid of it (with a few other fixes, please review again)